pkg/scheduler/preemption/preemption_test.go

```diff
@@ -1879,6 +1881,10 @@ func TestPreemption(t *testing.T) {
 				t.Errorf("Reported %d preemptions, want %d", preempted, tc.wantPreempted.Len())
 			}
 
+			if diff := cmp.Diff(beforeSnapshot, snapshotWorkingCopy, snapCmpOpts); diff != "" {
+				t.Errorf("Snapshot was modified (-initial,+end):\n%s", diff)
+			}
+
 			beforeResourceNodes := resourceNodes(beforeSnapshot)
```

Comment:
Please revert #4658. These diffs should be done as part of the snapshot comparison